### ISSUE TITLE
Stabilize existing syntax highlighting grammar

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/vscode-terraform#735]
TL;DR: Short term solution & preparation for https://github.com/hashicorp/vscode-terraform/issues/638

--- 

## Problem Statement

Users have reported a number of issues related to syntax highlighting in the past, some of which were still not yet triaged/confirmed or fixed, partially because the grammar is regex based which makes it prone to reintroduction of previous errors whenever edited without any test coverage.

## Expected User Experience

When user opens a `*.tf` or `*.tfvars` file in VS Code it is highlighted as expected and known inaccuracies which were reported (and reproduced) via GitHub are addressed and the grammar is tested against those.

## Proposal

 - Write tests for the grammar using https://github.com/microsoft/vscode-textmate
   - where possible tests should ideally be generic enough to test HCL syntax, _not_ Terraform language
   - this will help us ensure that transition towards a generic HCL grammar is smooth and we don't miss
 - Triage all `syntax` labelled issues and PRs & add any snippets from there as test cases
   - https://github.com/hashicorp/vscode-terraform/labels/syntax
   - https://github.com/hashicorp/vscode-terraform/pulls?q=is%3Apr+is%3Aopen+label%3Asyntax

**Comment by jpogran** [comment]
Test infrastructure added in https://github.com/hashicorp/vscode-terraform/pull/912

Test cases added in https://github.com/hashicorp/vscode-terraform/pull/914
